openapi.yaml

```diff
@@ -0,0 +1,2412 @@
+        - running
+        - failed
+
+    TriggerRule:
+      type: string
+      enum:
+        - all_success
+        - all_failed
+        - all_done
+        - one_success
+        - one_failed
+        - none_failed
+        - none_skipped
+        - none_failed_or_skipped
+        - dummy
+
+    WeightRule:
+      type: string
+      enum:
+        - downstream
+        - upstream
+        - absolute
+
+  # Reusable path, query, header and cookie parameters
+  parameters:
+    # Pagination parameters
+    PageOffset:
+      in: query
+      name: offset
+      required: false
+      schema:
+        type: integer
+        minimum: 0
+      description: The number of items to skip before starting to collect the result set.
+
+    PageLimit:
+      in: query
+      name: limit
+      required: false
+      schema:
+        type: integer
+        minimum: 1
+        default: 100
+      description: The numbers of items to return.
+
+    # Database entity fields
+    ConnectionID:
+      in: path
+      name: connection_id
+      schema:
+        type: integer
+      required: true
+      description: The Connection ID.
+
+    DAGID:
+      in: path
+      name: dag_id
+      schema:
+        type: string
+      required: true
+      description: The DAG ID.
+
+    TaskID:
+      in: path
+      name: task_id
+      schema:
+        type: string
+      required: true
+      description: The Task ID.
+
+    DAGRunID:
+      in: path
+      name: dag_run_id
+      schema:
+        type: string
+      required: true
+      description: The DAG Run ID.
+
+    TaskTryNumber:
+      in: path
+      name: task_try_number
+      schema:
+        type: integer
+      required: true
+      description: The Task Try Number.
+
+    EventLogID:
+      in: path
+      name: event_log_id
+      schema:
+        type: integer
+      required: true
+      description: The Event Log ID.
+
+    ImportErrorID:
+      in: path
+      name: import_error_id
+      schema:
+        type: integer
+      required: true
+      description: The Import Error ID.
+
+    PoolName:
+      in: path
+      name: pool_name
+      schema:
+        type: string
+      required: true
+      description: The Pool name.
+
+    VariableKey:
+      in: path
+      name: variable_key
+      schema:
+        type: string
+      required: true
+      description: The Variable Key.
+
+    # Logs
+    FullContent:
+      in: query
+      name: full_content
+      schema:
+        type: boolean
+      required: false
+      description: >
+        A full content will be returned.
+        By default, only the first fragment will be returned.
+
+    ContinuationToken:
+      in: query
+      name: token
+      schema:
+        type: string
+      required: false
+      description: >
+        A token that allows you to continue fetching logs.
+        If passed, it will specify the location from which the download should be continued.
+
+    XComKey:
+      in: path
+      name: key
```

Comment:
```suggestion
      name: key
```
```suggestion
      name: xcom_key
```

Reply:
I'm not a fan of this really --  what specifically is the problem this causes for Java?

Reply:
Here are a related change. (I did it before I read your message.) We can revert it if we make a different decision.
https://github.com/apache/airflow/pull/8721/commits/5993aef319918de0decc9bf0a0ed1379bd15a7ce

Reply:
Generated code of the method uses internally "key" variable to loop through the parameters. In Java you cannot have same variable and parameter name. 

I will be  happy if you find any other solution for that.

Reply:
Code here: https://github.com/mik-laj/airflow-api-clients/blob/master/out/object/java-google-api-client/src/main/java/org/openapitools/client/api/XComApi.java#L108 (now the parameter is named xcomkey - it used to be named key)

Reply:
And BTW. this is the main reason why I created https://github.com/apache/airflow/issues/9080 and I am super keen on automating the checks if the main language open-api clients can be automatically generated and compiled. Sometimes thing like choosing a name might matter (of course it should not matter in theory but we are leaving in an imperfect world). Making the clients generated and being usable is a priority for me in this case,

Reply:
> Code here: https://github.com/mik-laj/airflow-api-clients/blob/master/out/object/java-google-api-client/src/main/java/org/openapitools/client/api/XComApi.java#L108 (now the parameter is named xcomkey - it used to be named key)

It didn't cause any problems in my compile tests...

Ah. I tested with the java-okhttp-gson, and it doesn't have that problem.

So next question: do we want to/need support _every_ possible API generator? that's a lot of combinations to build+test+support?

Have we reported this bug upstream too?